MakeHist/HTLT.C

```diff
@@ -1,27 +1,51 @@
-void HTLT(){
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TFile.h>
+#include <TText.h>
+#include <TPaveText.h>
+#include <TLegend.h>
+#include <THStack.h>
+#include <TString.h>
+#include <iostream>
+
+void HTLT(TString prefix){
 
     THStack *hs = new THStack("hs","35.9 fb^{-1} (13TeV)");
 
-    //Signal input files
-    TFile *zp200 = new TFile("/cms/ldap_home/hyeahyun/signal/zp200/zp200_basic_DiMu_bjetpT20_Histograms.root");
-    TFile *zp350 = new TFile("/cms/ldap_home/hyeahyun/signal/zp350/zp350_basic_DiMu_bjetpT20_Histograms.root");
-    TFile *zp500 = new TFile("/cms/ldap_home/hyeahyun/signal/zp500/zp500_basic_DiMu_bjetpT20_Histograms.root");
-    
-    //Background input files
-    TFile *DY = new TFile("DY/DY_basic_DiMu_bjetpT20_Histograms.root");
-    TFile *TT = new TFile("TT/TT_basic_DiMu_bjetpT20_Histograms.root");
-    TFile *DB = new TFile("diboson/DB_basic_DiMu_bjetpT20_Histograms.root");
-    TFile *ST = new TFile("ST/ST_basic_DiMu_bjetpT20_Histograms.root");
+    //Open Files
+    TFile *zp200           = new TFile("./output/"+ prefix + "_zp200.root");
+    TFile *zp350           = new TFile("./output/"+ prefix + "_zp350.root");
+    TFile *zp500           = new TFile("./output/"+ prefix + "_zp500.root");
 
-    //Branch
+    TFile *WW              = new TFile("./output/"+ prefix + "_WWTo2L2Nu_13TeV.root");
+    TFile *ST_tW_antitop   = new TFile("./output/"+ prefix + "_ST_tW_antitop_5f_inclusiveDecays_13TeV.root");
+    TFile *TT              = new TFile("./output/"+ prefix + "_TTTo2L2Nu_TuneCUETP8M2_ttHtranche3_13TeV.root");
+    TFile *ST_tW_top       = new TFile("./output/"+ prefix + "_ST_tW_top_5f_inclusiveDecays_13TeV.root");
+    TFile *WZ              = new TFile("./output/"+ prefix + "_WZ_TuneCUETP8M1_13TeV.root");
+    TFile *ZZ              = new TFile("./output/"+ prefix + "_ZZ_TuneCUETP8M1_13TeV.root");
+    TFile *DY              = new TFile("./output/"+ prefix + "_DYJetsToLL_M.root");
+
+    //Make TH1Fs
     TH1F *HTLT_200 = (TH1F*)zp200->Get("HTLT_hist");
     TH1F *HTLT_350 = (TH1F*)zp350->Get("HTLT_hist");
     TH1F *HTLT_500 = (TH1F*)zp500->Get("HTLT_hist");
-    TH1F *HTLT_DY = (TH1F*)DY->Get("HTLT_hist");
+
+    //first is named DB as it will include ww, wz, and zz
+    TH1F *HTLT_DB = (TH1F*)WW->Get("HTLT_hist");
+    TH1F *HTLT_WZ = (TH1F*)WZ->Get("HTLT_hist");
```

Comment:
Can't you just concatenate these to e.g.
`HTLT_DB->Add((TH1F*)WW->Get("HTLT_hist"));`?
Two lines less.

Reply:
I'm getting an error:

"Error in <TH1F::Merge>: Cannot merge histograms - the histograms have different limits and undeflows/overflows are present. The initial histogram is now broken!"

Reply:
I just checked that this here works in the command shell:
`TH1F *test=new TH1F("test","bla",100,-1.,1.)
TH1F *test2=(TH1F*)test->Clone("test2")
test->FillRandom("gaus",1000)
test2->FillRandom("gaus",1000)
test2->Add((TH1F*)test->Clone("test3"))
(Bool_t) true
test2->Draw()`
and it has 2k entries. So this puzzles me.